### PR TITLE
Convert first test to use robot pattern

### PR DESCRIPTION
**Pull request by TGohil**
Converted the first test to use a robot pattern

### CODE REVIEW

File: app/src/androidTest/java/com/example/android/courtcounter/tests/CourtCounterTest.kt [hunk 1]
```diff
@@ -1,32 +1,47 @@
...
-
-        onView(allOf(withId(R.id.team_a_score), withText(TEAM_A_SCORE)))
-                .check(matches(isCompletelyDisplayed()))
+        with(counterTestRobot){
+            launchApp()
+        }
 
+        with(counterTestRobot) {
+            checkTeamAPointsIsDisplayed()
+        }
```
Comment (andrew-wrethman-yoti) [inline app/src/androidTest/java/com/example/android/courtcounter/tests/CourtCounterTest.kt:44]: So, the power of the with standard function is you can chain things like this together, you don't need two separate instances 
```
with(counterTestRobot) {
           launchApp()
           checkTeamAPointsIsDisplayed()
}
```

Interesting read on it - https://medium.com/@elye.project/mastering-kotlin-standard-functions-run-with-let-also-and-apply-9cd334b0ef84

File: app/src/androidTest/java/com/example/android/courtcounter/tests/CourtCounterTest.kt [hunk 2]
```diff
@@ -1,32 +1,47 @@
...
+import org.junit.Before
 import org.junit.Test
 import org.junit.runner.RunWith
 
 //type out the run with line and it will import the required parts
 @RunWith(AndroidJUnit4::class)
 class MyFirstTest: BaseTest() {
 
-    val TEAM_A_SCORE = "0"
+    private lateinit var counterTestRobot: CounterTestRobot
```
Comment (andrew-wrethman-yoti) [inline app/src/androidTest/java/com/example/android/courtcounter/tests/CourtCounterTest.kt:26]: So, you don't need to set this up as a lateinit var here. You can set this up like this 
`private val counterTestRobot = CounterTestRobot`

Comment (andrew-wrethman-yoti) [reply app/src/androidTest/java/com/example/android/courtcounter/tests/CourtCounterTest.kt:26]: The reason you may have seen this in the app is a lateinit var allows us to setup a var later on in a class, not where its first declared. We do this as we often rely on things like context/store which are only available after running super.setup

File: app/src/androidTest/java/com/example/android/courtcounter/robot/CounterTestRobot.kt [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    val TEAM_A_SCORE = "0"
+
+    fun launchApp() {
+
+        ActivityScenario.launch(MainActivity::class.java)
+    }
+
+    fun checkTeamAPointsIsDisplayed() {
+
+        onView(allOf(ViewMatchers.withId(R.id.team_a_score), withText(TEAM_A_SCORE)))
```
Comment (andrew-wrethman-yoti) [inline app/src/androidTest/java/com/example/android/courtcounter/robot/CounterTestRobot.kt:23]: You might notice(particularly if you are copying/pasting code) imports will be added back in, as we have with ViewMatcher. Whether or not you streamline this is up to you. Just something to keep an eye on

**Review by andrew-wrethman-yoti — COMMENTED** [review]
Nice work man!
Just a few things to consider

File: app/src/androidTest/java/com/example/android/courtcounter/robot/CounterTestRobot.kt [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import androidx.test.espresso.assertion.ViewAssertions
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import com.example.android.courtcounter.MainActivity
+import com.example.android.courtcounter.R
+import org.hamcrest.Matchers.allOf
+
+class CounterTestRobot {
+
+    val TEAM_A_SCORE = "0"
```
Comment (oz-saat) [inline app/src/androidTest/java/com/example/android/courtcounter/robot/CounterTestRobot.kt:14]: Not a big deal but as your test cases grow you might end up with a lot of constants floating around in different classes. To make things cleaner you can create a TestConstants class (see https://github.com/ozsaat/Fruity/blob/master/app/src/androidTest/java/com/osaat/fruity/utils/TestConstants.kt).

**Comment by oz-saat** [comment]
You've done really well here. It's looking very good.

**Pull request merged**